pkg/initializer/validator/datavalidator.go

```diff
@@ -90,16 +90,19 @@ func (d *DataValidator) sanityCheck(failBelowRevision int64) (DataDirStatus, err
 		// create the file `safe_guard` if it doesn't exist
 		if _, err := os.Stat(path); err != nil {
 			if errors.Is(err, os.ErrNotExist) {
-				data := []byte(namespace)
-				err := os.WriteFile(path, data, 0600)
-				if err != nil {
+				if err = os.WriteFile(path, []byte(namespace), 0600); err != nil {
 					d.Logger.Fatalf("can't create `safe_guard` file because : %v", err)
 				}
 			} else {
 				d.Logger.Fatalf("can't check if the `safe_guard` file exists or not because : %v", err)
 			}
 		}
 
+		// change file permission to handle previously created files with too wide permissions.
```

Comment:
can you also add a `TODO` comment to remove this additional condition in later release.

Reply:
Done